### PR TITLE
chore: add PR automation workflow for label management and status updates

### PR DESCRIPTION
**Pull request by jeromy-cannon**
## Description

This pull request consolidates and significantly expands the automation for pull request (PR) management in the repository's GitHub Actions workflows. The previous workflow for auto-assigning merge queue PRs is removed, and a new, more comprehensive workflow is introduced to handle PR assignment, labeling, and status management. The new workflow also automates several key processes for both regular and Dependabot PRs.

The most important changes are:

**Workflow consolidation and expansion:**
* The old `.github/workflows/zxf-auto-assign-merge-queue-prs.yaml` workflow is removed, and its functionality is merged into a new, broader workflow in `.github/workflows/zxf-prs-automation.yaml`. This new workflow covers a wider range of PR-related events and adds more automation features. [[1]](diffhunk://#diff-b91b3c96ec15debbf11ecb2af351b8f5a66cfebb746ee494fd74f937396fa4dcL1-L34) [[2]](diffhunk://#diff-80f191b35f91b5ba4fb76f3906013a26d05fe505bab025842d7946daa6980060R1-R267)

**Automated PR assignment and labeling:**
* PRs targeting `trunk-merge/*` branches are automatically assigned using the `auto-assign-action`, maintaining the previous auto-assignment behavior but now as part of the consolidated workflow.
* PRs with merge conflicts are automatically labeled (`PR: Merge Conflicts`) and converted to draft status; the label is removed and draft status reverted if conflicts are resolved.
* PRs with unresolved review threads are labeled (`PR: Unresolved Comments`), and the label is removed when all threads are resolved.
* If a PR is labeled as `Blocked` and is open, it is automatically converted to draft.

**Dependabot and priority label management:**
* Dependabot PRs are automatically assigned to the `swirlds-automation` user, default Dependabot labels are stripped, and a `P2-⏭️💎` priority label is added if none is set.
* For non-Dependabot PRs, if no priority label is present, the workflow checks linked issues (including those referenced in the PR body) and applies the highest priority label found among them.

**Event and schedule coverage:**
* The new workflow triggers on a wide

### Related Issues

* Closes #
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: .github/workflows/zxf-prs-automation.yaml [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                } else {
+                  await removeLabelIfPresent(prNumber, 'PR: Unresolved Comments', labels);
+                }
+
+                // Feature 3: Blocked label on an open PR → convert to draft.
+                if (labels.includes('Blocked') && isOpen) {
+                  await convertToDraft(pr.node_id, prNumber);
+                }
+
+                // Feature 5: Dependabot PRs → assign swirlds-automation, strip default labels, add P2.
```
Comment (jeromy-cannon) [inline .github/workflows/zxf-prs-automation.yaml]: Feature numbering is out of order — this is labeled `Feature 5` but runs before `Feature 4` (linked issues, line 232). Renumber so comments match execution order:

```suggestion
                // Feature 4: Dependabot PRs → assign swirlds-automation, strip default labels, add P2.
```

File: .github/workflows/zxf-prs-automation.yaml [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+                  for (const label of DEPENDABOT_DEFAULT_LABELS) {
+                    await removeLabelIfPresent(prNumber, label, labels);
+                  }
+                  const hasPriorityLabel = labels.some(l => PRIORITY_LABELS.includes(l));
+                  if (!hasPriorityLabel) {
+                    await addLabelIfMissing(prNumber, 'P2-⏭️💎', labels);
+                  }
+                }
+
+                // Feature 4: Linked issues → propagate highest priority label when none is set.
```
Comment (jeromy-cannon) [inline .github/workflows/zxf-prs-automation.yaml]:
```suggestion
                // Feature 5: Linked issues → propagate highest priority label when none is set.
```

File: .github/workflows/zxf-prs-automation.yaml [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+                core.warning(`Failed to convert PR #${prNumber} to draft: ${error.message}`);
+              }
+            }
+
+            async function checkUnresolvedThreads(prNumber) {
+              const result = await github.graphql(`
+                query($owner: String!, $repo: String!, $number: Int!) {
+                  repository(owner: $owner, name: $repo) {
+                    pullRequest(number: $number) {
+                      reviewThreads(first: 100) {
```
Comment (jeromy-cannon) [inline .github/workflows/zxf-prs-automation.yaml]: `first: 100` is the page ceiling — a PR with >100 review threads would return an incomplete set and could incorrectly report no unresolved threads. Consider checking `pageInfo.hasNextPage` and fetching subsequent pages.

File: .github/workflows/zxf-prs-automation.yaml [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+                    }
+                  }
+                }
+              `, { owner, repo, number: prNumber });
+
+              const closingIssues = result.repository.pullRequest.closingIssuesReferences.nodes;
+              const closingNumbers = new Set(closingIssues.map(i => i.number));
+
+              // Parse body for "related to #N" patterns not already covered by closing references.
+              const relatedPattern = /(?:related\s+to|relates?\s+to|connected?\s+to|see\s+also)\s+#(\d+)/gi;
```
Comment (jeromy-cannon) [inline .github/workflows/zxf-prs-automation.yaml]: `connected?` makes the `d` optional, matching “connecte to” unintentionally. Use a non-capturing group:

```suggestion
              const relatedPattern = /(?:related\s+to|relates?\s+to|connect(?:ed)?\s+to|see\s+also)\s+#(\d+)/gi;
```

File: .github/workflows/zxf-prs-automation.yaml [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+              if (prNumber) {
+                await processPR(prNumber);
+              } else {
+                core.warning('Could not determine PR number from event payload');
+              }
+            } else {
+              // push / schedule / workflow_dispatch — scan all open PRs.
+              const { data: openPRs } = await github.rest.pulls.list({
+                owner, repo, state: 'open', per_page: 100,
+              });
```
Comment (jeromy-cannon) [inline .github/workflows/zxf-prs-automation.yaml]: `per_page: 100` is the GitHub API maximum — any open PRs beyond the first 100 are silently skipped on scheduled/push runs. Fine in practice if the repo rarely exceeds 100 open PRs, but worth paginating if that ever changes.

**Review by jan-milenkov — DISMISSED** [review]

**Review by jan-milenkov — APPROVED** [review]
